lightning/src/onion_message/functional_tests.rs

```diff
@@ -112,16 +112,31 @@ impl Writeable for TestCustomMessage {
 
 struct TestCustomMessageHandler {
 	expected_messages: Mutex<VecDeque<TestCustomMessage>>,
```

Comment:
I wonder if we should just make this hold a `(TestCustomMessage, bool)` and add an `expect_message_and_respond` method instead of adding `include_reply_path`. Then you don't need to reset the `bool` after handling a message.

Reply:
Thanks a lot for the idea!
Built further upon the idea after offline discussion.

1. Structured the expectations in the `Expectation` struct.
2. Introduce the `expect_message_and_response` function in [594dc0f](https://github.com/lightningdevkit/rust-lightning/commit/594dc0fd893eb85abde3ee7cc8527b225b042cb4)